### DISCUSSION TITLE
List of all default properties

### DISCUSSION THREAD

**Discussion by ramsbaron** [getgauge/gauge#1883] [General]
What is the default value of gauge_screenshots_dir?

for that matter, is there a list of all the various environment properties supported by gauge, and their default values?

**Comment by sriv** [comment]
> What is the default value of gauge_screenshots_dir?

default value of `gauge_screenshots_dir` is `<project_root>/.gauge/screenshots`. 

> for that matter, is there a list of all the various environment properties supported by gauge, and their default values?

Most of the configuration is listed in https://docs.gauge.org/configuration.html. This one is missing, a PR would be nice.

**Reply by ramsbaron** [reply]
Thanks @sriv
You are right that most of the configuration is listed in https://docs.gauge.org/configuration.html. However, I did not find the `gauge_screenshots_dir` property in that file (on my installation) either. It makes me think there are properties that are in play even if they are not in any `.properties` file. My intent is to find a list of those _hidden_ properties.

**Reply by zabil** [reply]
I think the best place to look for this is the source code for e.g.

https://github.com/getgauge/gauge/blob/master/env/env.go
https://github.com/getgauge/gauge/blob/master/config/configuration.go

Default values are at
https://github.com/getgauge/gauge/blob/3d4244430dc40b76181874f1d00cbb24e90a52e4/env/env.go#L103-L117

**Reply by ramsbaron** [reply]
fantastic, thanks a ton @zabil
